csharp/src/Apache.Arrow/Field.Builder.cs

```diff
@@ -52,6 +55,131 @@ public Builder DataType(IArrowType type)
+                    // Binary
+                    case var _ when valueType == typeof(byte):
+                        DataType(new BinaryType());
+                        break;
+                    case var _ when valueType == typeof(IEnumerable<byte>):
+                        DataType(new BinaryType());
+                        Nullable(true);
+                        break;
+                    // Boolean
+                    case var _ when valueType == typeof(bool):
+                        DataType(new BooleanType());
+                        break;
+                    // String
+                    case var _ when valueType == typeof(string):
+                        DataType(new StringType());
+                        Nullable(true);
+                        break;
+                    // Integers
+                    case var _ when valueType == typeof(sbyte):
+                        DataType(new Int8Type());
+                        break;
+                    case var _ when valueType == typeof(short):
+                        DataType(new Int16Type());
+                        break;
+                    case var _ when valueType == typeof(int):
+                        DataType(new Int32Type());
+                        break;
+                    case var _ when valueType == typeof(long):
+                        DataType(new Int64Type());
+                        break;
+                    // Unisgned Integers
+                    case var _ when valueType == typeof(ushort):
+                        DataType(new UInt16Type());
+                        break;
+                    case var _ when valueType == typeof(uint):
+                        DataType(new UInt32Type());
+                        break;
+                    case var _ when valueType == typeof(ulong):
+                        DataType(new UInt64Type());
+                        break;
+                    // Decimal
+                    case var _ when valueType == typeof(decimal):
+                        DataType(Decimal256Type.Default);
+                        break;
+                    // Float
+                    case var _ when valueType == typeof(float):
+                        DataType(new FloatType());
+                        break;
+                    // Double
+                    case var _ when valueType == typeof(double):
+                        DataType(new DoubleType());
+                        break;
+                    // DateTime
+                    case var _ when valueType == typeof(DateTime) || valueType == typeof(DateTimeOffset):
+                        DataType(new TimestampType(TimeUnit.Nanosecond, timezone));
+                        break;
+                    // Time
+                    case var _ when valueType == typeof(TimeSpan):
+                        DataType(new Time64Type(TimeUnit.Nanosecond));
+                        break;
+#if NETCOREAPP3_1_OR_GREATER
+                    // Dictionary: IDictionary<?,?>
```

Comment:
I think it would be very unusual that a dictionary encoded type is represented with a .NET Dictionary.  I'm pretty sure they are two separate things.  A .NET Dictionary would probably be closer to `MapArray` but I don't see that in the .NET impl currently.

Reply:
indeed, will need to implement MapType